README.md

```diff
@@ -50,6 +50,68 @@ Maximum upload size can be specified in project settings under `MAX_FILE_UPLOAD_
 
 `django.db.models.CharField` that stores `NULL` but returns ''.
 
+### 3. SmartQuerySet `chamber.models.SmartQuerySet`
+SmartModel introduced to Chamber uses by default a modified QuerySet with some convenience filters. 
+
+If you are overriding model manager of a SmartModel, you should incorporate `SmartQuerySet` in order not to lose its benefits and to follow the Rule of the Least Surprise (everyone using your SmartModel will assume the custom filters to be there).
+
+1. If the manager is created using the `QuerySet.as_manager()` method, your custom queryset should subclass `SmartQuerySet` instead the one from Django.
+2. If you have a new manager created by subclassing `models.Manager` from Django, you should override the `get_queryset` method as shown in Django docs [here](https://docs.djangoproject.com/en/1.10/topics/db/managers/#calling-custom-queryset-methods-from-the-manager).
+
+List of the added filters follows.
+
+#### 3.1 `fast_distinct()`
+Returns same result as regular `distinct()` but is much faster especially in PostgreSQL which performs distinct on all DB columns. The optimization is achieved by doing a second query and the `__in` operator. If you have queryset `qs` of `MyModel` then `fast_distinct()` equals to calling
+```python
+MyModel.objects.filter(pk__in=qs.values_list('pk', flat=True))
+```
+
+
+### 4. Model Dispatchers
+
+Model dispatchers are a way to reduce complexity of `_pre_save` and `_post_save` methods of the SmartModel. A common use-case of these methods is to perform some action based on the change of the model, e.g. send a notification e-mail when the state of the invoice changes.
+
+Better alternative is to define a handler function encapsulating the action that should happen when the model changes in a certain way. This handler is registered on the model using a proper dispatcher.
+
+So far, there are two types of dispatchers but you are free to subclass the `BaseDispatcher` class to create your own.
+
+The moment when the handler should be fired may be important. Therefore, you can register the dispatcher either in the `pre_save_dispatchers` group or `post_save_dispatchers` group. Both groups are dispatched immediately after the `_pre_save` or `_post_save` method respectively.
+
+#### 4.1 Property Dispatcher
+`chamber.models.dispatchers.StateDispatcher` is a versatile dispatcher that fires the given handler when a specified property of the model evaluates to `True`.
+
+The example shows how to to register `send_email` handler to be dispatched after saving the object if the property `should_send_email` returns `True`.
+```python
+class MySmartModel(chamber_models.SmartModel):
+    
+    email_sent = models.BooleanField()
+
+    post_save_dispatchers = (
+        PropertyDispatcher(send_email, 'should_send_email'),
```

Comment:
Já vím, proč jsem to neudělal -- ono to totiž nejde. Může to přijmout string nebo funkci, ale property se nedá zavolat, navíc na ní nikde nevidím, že by byla bound na ten self, je to prostě objekt plující v paměti a nemůžeš k němu přistoupit. Musíš udělat `getattr(self, 'my_property')`, protože to je metoda, která simuluje atribut objektu (a dělá to správně).

Takže bych to nechal, jak to je. Nechci, aby to přijímalo funkce, protože to nemá jak získat nějaké další argumenty (leda to poslat jako kwargs do save metody, ale to je prostě jiný dispatcher, nejde to míchat s tímhle).